server/swap/swap.go

```diff
@@ -347,26 +403,88 @@ func (s *Swapper) UserSwappingAmt(user account.AccountID, base, quote uint32) (a
 	return
 }
 
+// pendingAccountStats is used to sum in-process match stats for the
+// AccountStats method.
+type pendingAccountStats struct {
+	acctAddr string
+	assetID  uint32
+	swaps    uint64
+	qty      uint64
+	redeems  int
+}
+
+func newPendingAccountStats(acctAddr string, assetID uint32) *pendingAccountStats {
+	return &pendingAccountStats{
+		acctAddr: acctAddr,
+		assetID:  assetID,
+	}
+}
+
+func (p *pendingAccountStats) addMatch(mt *matchTracker) {
+	p.addOrder(mt, mt.Maker, order.MakerSwapCast, order.MakerRedeemed)
+	p.addOrder(mt, mt.Taker, order.TakerSwapCast, order.MatchComplete)
+}
+
+func (p *pendingAccountStats) addOrder(mt *matchTracker, ord order.Order, swappedStatus, redeemedStatus order.MatchStatus) {
+	trade := ord.Trade()
+	if ord.Base() == p.assetID && trade.BaseAccount() == p.acctAddr {
+		if trade.Sell {
+			if mt.Status < swappedStatus {
+				p.qty += mt.Quantity
+				p.swaps++
+			}
+		} else if mt.Status < redeemedStatus {
+			p.redeems++
+		}
+	}
+	if ord.Quote() == p.assetID && trade.QuoteAccount() == p.acctAddr {
+		if !trade.Sell {
+			if mt.Status < swappedStatus {
+				p.qty += calc.BaseToQuote(mt.Rate, mt.Quantity)
+				p.swaps++ // The swap is expected to occur in 1 transaction.
+			}
+		} else if mt.Status < redeemedStatus {
+			p.redeems++
+		}
+	}
+}
+
+// AccountStats is part of the MatchNegotiator interface to report in-process
+// match information for a asset account address.
+func (s *Swapper) AccountStats(acctAddr string, assetID uint32) (qty, swaps uint64, redeems int) {
+	stats := newPendingAccountStats(acctAddr, assetID)
```

Comment:
So these three stats values returned describe remaining transactions?  As in `redeems` is a count of redeem transactions that are expected to still take place for active matches, and similar for `swaps` (could that be `inits` to clarify?).
This part is reasonable clear from the `AccountStats interface` docs, but `qty` could be clearer there imo.  `qty` is the amount that will leave this asset-account when said inits/swaps are sent.